cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -310,7 +310,9 @@ public OperationSnapshot createCluster(String name, String imageVersion, Map<Str
     FirewallList firewalls = compute.firewalls().list(projectId).execute();
     List<String> tags = new ArrayList<>();
     Set<FirewallPort> requiredPorts = EnumSet.allOf(FirewallPort.class);
-    for (Firewall firewall : firewalls.getItems()) {
+
+    // Iterate all firewall rules and see if it has ingress rules for all required firewall port.
+    for (Firewall firewall : Optional.ofNullable(firewalls.getItems()).orElse(Collections.emptyList())) {
```

Comment:
in the case of empty firewall rules list, won't line 349 throw an error?

Reply:
Yes. SSH ingress is by default not allowed.